assignment-1/cause_2.md

```diff
@@ -0,0 +1,68 @@
+# Architecture - rigid and lacking non-functional requirements
+
+## Key factors in failure
+
+One of the primary problems with HealthCare.gov was its performance.
+The architecture was designed to pass on requests from the web-servers real-time to the back-end systems.
+The system was not designed to be able to handle the high loads that it faced during the go-live [2].
+Late during development, they discovered that the new requirements would need changes in the architecture.
+However, the current architecture did not provide the flexibility to allow such changes in it.
+
+Another one of the problems the site faced, was that there were a number of security vulnerabilities [6].
+These came to light during a security breach in the July of 2014. The breach was discovered by a CMS security team in the following month. It occurred due to a development server not being properly configured and this led to malware being uploaded to the system. 
+
+
+##  Software development methodologies
+
+Could the problems stated above have been fixed by using software methodologies?
+
+
+### Scrum
+
+The back-end of HealthCare.gov is reported to have been built using agile practices but they were likely not applied correctly due to the project having explicit phases defined such as testing [1].
```

Comment:
Scrum prescribes cross-functional teams: "Cross-functional teams have all competencies needed to accomplish the work without depending on others not part of the team. "
Scrum prescribes a "definition of done". If testing is required for a releasable increment, it should be in this definition and be done within the sprint. Scrum does not include phases, such as in RUP.